resharper/resharper-unity/src/Rider/RiderPackageUpdateAvailabilityChecker.cs

```diff
@@ -0,0 +1,127 @@
+        {
+            if (!myBoundSettingsStore.GetValue((UnitySettings s) => s.AllowRiderUpdateNotifications))
+                return;
+            
+            myPackageManager.IsInitialUpdateFinished.WhenTrue(lifetime, lt =>
+            {
+                var version = myUnityVersion.ActualVersionForSolution.Value;
+                if (version == null)
+                    return;
+            
+                if (myNotificationShown.Contains(mySolution.SolutionFilePath)) return;
+
+                // Version before 2019.2 doesn't have Rider package
+                // 2019.2.0 - 2019.2.5 : version 1.2.1 is the last one
+                // 2019.2.6 - present : 3.0.7+
+                if (version < new Version(2019, 2, 6)) return;
+
+                var package = myPackageManager.GetPackageById(packageId);
+
+                if (package == null)
+                {
+                    myNotificationShown.Add(mySolution.SolutionFilePath);
+                    myLogger.Info($"{packageId} is missing.");
+                    var notification = new NotificationModel($"JetBrains Rider package in Unity is missing.", "Make sure JetBrains Rider package is installed in Unity Package Manager.", true, RdNotificationEntryType.WARN, new List<NotificationHyperlink>());
+                    myShellLocks.ExecuteOrQueueEx(lt, "RiderPackageUpdateAvailabilityChecker.ShowNotificationIfNeeded", () => myNotifications.Notification(notification));
+                }
+                // todo: read available newest compatible Rider package version from somewhere, like protocol or json file
+                else if (package.Source == PackageSource.Registry && new Version(package.PackageDetails.Version) < new Version(3,0,7))
+                {
+                    myNotificationShown.Add(mySolution.SolutionFilePath);
+                    myLogger.Info($"{packageId} {package.PackageDetails.Version} is older then expected.");
+                    var neverShowLink = new NotificationHyperlink("Never show for this solution", myHyperlinkId);
+                    var notificationId = 123409;
+                    var notification = new NotificationModel("Update available - JetBrains Rider package.", 
+                        "Check for JetBrains Rider package updates in Unity Package Manager.", 
+                        true, RdNotificationEntryType.INFO, new List<NotificationHyperlink> {neverShowLink}, notificationId);
+                    var notificationLifetime = lt.CreateNested();
+                    myNotifications.ExecuteHyperlink.Advise(notificationLifetime.Lifetime, id =>
+                    {
+                        if (id == myHyperlinkId)
+                        {
+                            mySettingsStore.BindToContextTransient(ContextRange.ManuallyRestrictWritesToOneContext(mySolution.ToDataContext()))
+                                .SetValue((UnitySettings key) => key.AllowRiderUpdateNotifications, false);
```

Comment:
This writes to `.sln.DotSettings`, doesn't it? Might be better to use the persistent solution properties, which are stored in the ReSharper cache folder, rather than the user's settings.

Reply:
It goes to .sln.dotSettings.user

Reply:
Please store that in solution caches, do not modify anything in project on openning

Reply:
If I start using persistent solution properties, how user would enable it back if occasionally turn off with the link in the popup? In the current implementation, I have a setting with a checkbox in the Preferences.